Dockerfile

```diff
@@ -1,31 +1,29 @@
-# Base
-ARG REGISTRY=562955126301.dkr.ecr.eu-west-2.amazonaws.com
+ARG PARENT_VERSION=latest
+ARG PARENT_REGISTRY=562955126301.dkr.ecr.eu-west-2.amazonaws.com
 
 # Development
-# FROM $REGISTRY/ffc-dotnet-parent/sdk:$BASE_VERSION AS 
-FROM dotnet-sdk-parent AS development
-RUN mkdir -p /app/FFCDemoPaymentService/
-WORKDIR /app/FFCDemoPaymentService/
-COPY --chown=www-data:www-data ./FFCDemoPaymentService/*.csproj ./
-RUN dotnet restore
-COPY --chown=www-data:www-data ./FFCDemoPaymentService/ ./
-RUN dotnet publish -c Release -o /app/out
-CMD ["--urls", "http://*:3007"]
-
-# Test
-FROM developent AS test
-RUN mkdir -p /app/FFCDemoPaymentService.Tests/
-WORKDIR /app/FFCDemoPaymentService.Tests/
-COPY --chown=www-data:www-data ./FFCDemoPaymentService.Tests/*.csproj ./
-RUN dotnet restore
-COPY --chown=www-data:www-data ./FFCDemoPaymentService.Tests/ ./
-ENTRYPOINT [ "dotnet", "test" ]
+FROM $PARENT_REGISTRY/ffc-dotnet-sdk:$PARENT_VERSION AS development
+ARG PARENT_VERSION
+ARG PARENT_REGISTRY
+ARG PORT=3007
+LABEL uk.gov.defra.ffc.parent-image=${PARENT_REGISTRY}/ffc-node-development:${PARENT_VERSION}
+RUN mkdir -p /home/dotnet/FFCDemoPaymentService/ /home/dotnet/FFCDemoPaymentService.Tests/
+COPY --chown=dotnet:dotnet ./FFCDemoPaymentService.Tests/*.csproj ./FFCDemoPaymentService.Tests/
+RUN dotnet restore ./FFCDemoPaymentService.Tests/FFCDemoPaymentService.Tests.csproj
+COPY --chown=dotnet:dotnet ./FFCDemoPaymentService/*.csproj ./FFCDemoPaymentService/
+RUN dotnet restore ./FFCDemoPaymentService/FFCDemoPaymentService.csproj
+COPY --chown=dotnet:dotnet ./FFCDemoPaymentService.Tests/ ./FFCDemoPaymentService.Tests/
+COPY --chown=dotnet:dotnet ./FFCDemoPaymentService/ ./FFCDemoPaymentService/
+RUN dotnet publish ./FFCDemoPaymentService/ -c Release -o /home/dotnet/out
+CMD [ "watch", "--project", "./FFCDemoPaymentService", "run", "--urls", "http://*:${PORT}" ]
```

Comment:
This line causes the application to fail when running due the the port variable.  The kestrel server fails to bind using the environment variable but works fine if you hard code 3007 here.  I don't think the variable is being substituted here and it's using the raw text.

Reply:
Glad you tested that! I was having trouble with Docker yesterday so I hadn't managed to test this and then forgot all about it. I'd done two things wrong there:

1. I didn't set `PORT` as an environment variable so it wasn't available at runtime
2. Using the [exec form](https://docs.docker.com/engine/reference/builder/#exec-form-entrypoint-example) of `ENTRYPOINT` and `CMD` (square-bracket notation) means they run without a shell environment, so can't access variables even if they are set.

I've switched to setting the `ENTRYPOINT` with a [shell form](https://docs.docker.com/engine/reference/builder/#shell-form-entrypoint-example) command (looks like a plain string) so it can pick up an environment variable, which is now set for `PORT`.

One downside of the shell-form is that you can't extend `ENTRYPOINT` with `CMD` anymore - it treats the entire CMD as a single argument to `sh -c` on the end of the `ENTRYPOINT`. e.g.
```
ENTRYPOINT dotnet
CMD watch --project ./FFCDemoPaymentService run --urls http://*:${PORT}
```
is equivalent to
```
ENTRYPOINT dotnet sh -c "watch --project ./FFCDemoPaymentService run --urls http://*:${PORT}"
```
That's why I've now switched to just setting ENTRYPOINT, rather than using CMD as well.